src/simulation_lifecycle/utils/file.c

```diff
@@ -18,4 +20,19 @@ int write_data_to_file(char *filepath, char *data) {
         }
     }
     return res;
+}
+
+void read_json_file(char *file_path, cJSON **p_target) {
+    FILE *f = fopen(file_path, "rb");
+    fseek(f, 0, SEEK_END);
+    long fsize = ftell(f);
+    fseek(f, 0, SEEK_SET);  /* same as rewind(f); */
+
+    char *string = malloc(fsize + 1);
+    fread(string, 1, fsize, f);
+    fclose(f);
+    string[fsize] = '\0';
+
+    *p_target = cJSON_Parse(string);
+    free(string);
```

Comment:
I think this is originally Roman's function, so we'll have three to merge it seems like.

Reply:
Don't worry, I'll deal with conflicts so only one remains